### ISSUE TITLE
Bug: Tracing issues

### ISSUE THREAD

**Issue by royassis** [awslabs/aws-lambda-powertools-python#4045]
### Expected Behaviour

Traceback is not being logged or raised

### Current Behaviour

Traceback attached below is being logged and raised, when the pymongo package and tracing is being used in code.

### Code snippet

```python
Traceback (most recent call last):
  File "/var/task/pymongo/monitoring.py", line 1519, in publish_command_success
    subscriber.succeeded(event)
  File "/var/task/aws_xray_sdk/ext/pymongo/patch.py", line 37, in succeeded
    subsegment.put_annotation('mongodb_duration_micros', event.duration_micros)
AttributeError: 'NoneType' object has no attribute 'put_annotation'
[WARNING]	2024-03-31T16:26:55.661Z	9a41172c-e61b-4d05-a460-6eed564ec3f7	Subsegment appdata@172.30.3.165:27017 discarded due to Lambda worker still initializing
Traceback (most recent call last):
  File "/var/task/pymongo/monitoring.py", line 1519, in publish_command_success
    subscriber.succeeded(event)
  File "/var/task/aws_xray_sdk/ext/pymongo/patch.py", line 37, in succeeded
    subsegment.put_annotation('mongodb_duration_micros', event.duration_micros)
```

### Steps to Reproduce

Create a function and initialize connection to mongodb database with pymongo.
Use tracer.capture_lambda_handler and tracer.capture_method.

Where:
```python
from aws_lambda_powertools import Logger, Tracer
tracer = Tracer().
```

### Powertools for AWS Lambda (Python) version

latest

### AWS Lambda function runtime

3.11

### Packaging format used

PyPi

**Comment by leandrodamascena** [comment]
Hey @royassis! It's great to see you back here! 😊

After examining the exception and reviewing the code, it appears that the error originates from [this line](https://github.com/aws/aws-xray-sdk-python/blob/master/aws_xray_sdk/ext/pymongo/patch.py#L36). I have a hypothesis regarding the cause, and I'd like to confirm it with you.

1 - Are you running this code locally? I'm asking because I noticed the presence of the `[WARNING]	2024-03-31T16:26:55.661Z	9a41172c-e61b-4d05-a460-6eed564ec3f7	Subsegment appdata@172.30.3.165:27017 discarded due to Lambda worker still initializing` line, which typically indicates an error when running outside of AWS Lambda. 
If this is the case, the error is expected. When running locally with an emulator, there is no X-Ray root segment, and therefore X-Ray cannot obtain the subsegment using `xray_recorder.current_subsegment()`. 
Just a quick explanation on this matter: when you activate "X-Ray active tracing" in your Lambda configuration on AWS, then AWS Lambda automatically generates a root segment. All subsequent Tracer information is then added in the form of subsegments.

2 - Are you using a Powertools version prior to 2.36.0? Starting from version 2.36.0, we automatically disable the X-Ray tracer in environments such as Chalice and AWS SAM to prevent this type of error.

I used this code in my Lambda function and it worked as expected:

```python
from aws_lambda_powertools import Tracer
from aws_lambda_powertools.utilities.typing import LambdaContext
from pymongo import MongoClient

tracer = Tracer() 

@tracer.capture_method
def db_conn():
    uri = "mongodb+srv://USER:PASS@test-powertools.xxx.mongodb.net/?retryWrites=true&w=majority&appName=test-powertools"
    mongodb_client = MongoClient(uri)
    return mongodb_client


@tracer.capture_lambda_handler
def lambda_handler(event: dict, context: LambdaContext) -> str:
    conn = db_conn()
    print(conn.server_info())
    return {"test": "ok"}
```

![image](https://github.com/aws-powertools/powertools-lambda-python/assets/4295173/49977374-13fa-4733-a55a-183953321039)

Thanks
